Casks/t/texstudio.rb

```diff
@@ -1,16 +1,21 @@
 cask "texstudio" do
+  arch arm: "-m1"
+  ext = on_arch_conditional arm: "zip", intel: "dmg"
+  app_suffix = on_arch_conditional arm: "-#{version}-osx-m1", intel: ""
+
   version "4.7.3"
```

Comment:
```suggestion
  version "4.7.3"
  
  ext = on_arch_conditional arm: "zip", intel: "dmg"
  app_suffix = on_arch_conditional arm: "-#{version}-osx-m1", intel: ""
```

Exchange the orders of 'version' and 'app_suffix' to fix the undefined variable.

Reply:
Thank you for your review. However, defining conditional statements after `version` is not permitted by stanza order (executing `brew style texstudio` after applying your patch shows this), so I think they cannot include version information.

Reply:
Can you update it to use `on_arch`  blocks instead?

Reply:
So, is it okay to revert this commit?

https://github.com/Homebrew/homebrew-cask/pull/169292/commits/7eefae7543c06e89f5980a7f7b2b61b418f7b754

Reply:
Because the `urls` differ so much, putting the entire `url` in the `on_{arch}` blocks would make the most sense to me.

Reply:
Hmm, I agree that the `url` is complicated, but since it works correctly, I thought it was the `app` that should use the `on_{arch}` block.